__tests__/app.test.js

```diff
@@ -170,6 +167,77 @@ describe('GET /api/articles/:article_id/comments', () => {
     });
 });
 
+describe('POST /api/articles/:article_id/comments', () => {
+    test('should return a 201 and the posted comment on success', () => {
+        const newComment = {
+            body: "This morning, I showered for nine minutes.", 
+            username: "butter_bridge"}
+        return request(app)
+        .post('/api/articles/1/comments')
+        .expect(201)
+        .send(newComment)
+        .then(({body}) => {
+                expect(body.comment).toMatchObject({
+                    comment_id: 19,
+                    votes: expect.any(Number),
+                    created_at: expect.any(String),
+                    article_id: 1,
+                    author: expect.any(String),
+                    body: expect.any(String)
+                })
+        })
+    });
+    test('should return a 400 Invalid Input if given an invalid id', () => {
+        const newComment = {
+            body: "This morning, I showered for nine minutes.", 
+            username: "butter_bridge"}
+        return request(app)
+        .post('/api/articles/invalid/comments')
+        .expect(400)
+        .send(newComment)
+        .then(({body}) => {
+            expect(body.msg).toBe('Invalid Input')
+        })
+    });
+    test('should return a 404 Article Does Not Exist if the id can\'t be found', () => {
+        const newComment = {
+            body: "This morning, I showered for nine minutes.", 
+            username: "butter_bridge"}
+        return request(app)
+        .post('/api/articles/20/comments')
+        .expect(404)
+        .send(newComment)
+        .then(({body}) => {
+            expect(body.msg).toBe('Article Does Not Exist')
+        })
+    });
+    test('should return a 400 Invalid Request Body when the request body is missing the body or username', () => {
+        const newComment = {
+            body: "This morning, I showered for nine minutes.", 
+            username: "butter_bridge",
+            extraKey: "extra key"}
+        return request(app)
+        .post('/api/articles/1/comments')
+        .expect(400)
+        .send(newComment)
+        .then(({body}) => {
+            expect(body.msg).toBe('Invalid Request Body')
+        })
```

Comment:
It's good that you've handled the extra key, but it can simply be ignored if there're extra properties